File: PrinciplesOfProgramming/Week11/mxt-9581/mxt-9581.py
def cal(x, n):
    if n == 1:
        return x / (1 + x)
    else:
        return x / (n + cal(x, n - 1))
    
value = input().split()
x = int(value[0])
n = int(value[1])

print(f"{cal(x, n):.2f}")

File: PrinciplesOfProgramming/Week11/mxt-9581/mxt-9581.c
#include <stdio.h>

double cal(double x, double n);

int main(void)
{
    double x, n;
    scanf("%lf %lf", &x, &n);

    printf("%.2lf", cal(x, n));
}

double cal(double x, double n)
{
    if (n == 1)
    {
        return x / (1 + x);
    }

    else
    {
        return x / (n + cal(x, n - 1));
    }
}